lib/jnpr/junos/utils/start_shell.py

```diff
@@ -21,28 +22,39 @@ def _ssh_exec(self, command):
 
     """
 
-    def __init__(self, nc):
+    def __init__(self, nc, timeout=30):
         """
         Utility Constructor
 
         :param Device nc: The Device object
+
+        :param int timeout:
+          Timeout value in seconds to wait for expected string/pattern.
         """
         self._nc = nc
+        self.timeout = timeout
```

Comment:
As mentioned in above case nc.timeout will throw exception

```
  File "xxx", line 17, in <module>
    with StartShell(dev) as ss:
  File "xxx", line 36, in __init__
    self.timeout = timeout or nc.timeout
  File "xxx", line 164, in timeout
    return self._conn.timeout
AttributeError: 'NoneType' object has no attribute 'timeout'
```

device.timeout is only being set when connection is made using dev.open()
Should we change this?